streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -152,25 +153,29 @@
                                         1,
                                         Importance.MEDIUM,
                                         REPLICATION_FACTOR_DOC)
-                                .define(KEY_SERIALIZER_CLASS_CONFIG,        // required with no default value
+                                .define(KEY_SERIALIZER_CLASS_CONFIG,
```

Comment:
I'd much prefer us to consolidate the `KEY_SERIALIZER_CLASS_CONFIG` and the `KEY_DESERIALIZER_CLASS_CONFIG` functionality into sth like a single `KEY_SERDE_CLASS_CONFIG`.  Again, the purpose of this PR is to unify Sers and Desers into a single Serde interface, and we don't want to continue requiring the user to specify Sers and Desers separately.

So, as a user, I would expect sth like:

```java
.define(KEY_SERDE_CLASS_CONFIG,
    Type.CLASS,
    ByteArraySerDe.class,
    Importance.HIGH,
    StreamsConfig.KEY_SERDE_CLASS_DOC) /* We can't use `ProducerConfig.*` here because the latter is only for serializers, not for dual-purpose `Serde`)
```